code/modules/admin/create_mob.dm

```diff
@@ -11,35 +11,29 @@
 	user << browse(create_panel_helper(create_mob_html), "window=create_mob;size=425x475")
 
 /proc/randomize_human(mob/living/carbon/human/human)
-	human.gender = pick(MALE, FEMALE)
+	if(human.dna.species.sexes)
+		human.gender = pick(MALE, FEMALE, PLURAL)
+	else
+		human.gender = PLURAL
 	human.physique = human.gender
 	human.real_name = human.dna?.species.random_name(human.gender) || random_unique_name(human.gender)
 	human.name = human.real_name
-	human.underwear = random_underwear(human.gender)
-	human.underwear_color = "#[random_color()]"
-	human.skin_tone = random_skin_tone()
 	human.hairstyle = random_hairstyle(human.gender)
 	human.facial_hairstyle = random_facial_hairstyle(human.gender)
 	human.hair_color = "#[random_color()]"
 	human.facial_hair_color = human.hair_color
 	var/random_eye_color = random_eye_color()
 	human.eye_color_left = random_eye_color
 	human.eye_color_right = random_eye_color
-	human.dna.blood_type = random_blood_type()
 
-	// Mutant randomizing, doesn't affect the mob appearance unless it's the specific mutant.
+	human.dna.blood_type = random_blood_type()
 	human.dna.features["mcolor"] = "#[random_color()]"
-	human.dna.features["ethcolor"] = GLOB.color_list_ethereal[pick(GLOB.color_list_ethereal)]
-	human.dna.features["tail_cat"] = pick(GLOB.tails_list_human)
-	human.dna.features["tail_lizard"] = pick(GLOB.tails_list_lizard)
-	human.dna.features["snout"] = pick(GLOB.snouts_list)
-	human.dna.features["horns"] = pick(GLOB.horns_list)
-	human.dna.features["frills"] = pick(GLOB.frills_list)
-	human.dna.features["spines"] = pick(GLOB.spines_list)
-	human.dna.features["body_markings"] = pick(GLOB.body_markings_list)
-	human.dna.features["moth_wings"] = pick(GLOB.moth_wings_list)
-	human.dna.features["moth_antennae"] = pick(GLOB.moth_antennae_list)
-	human.dna.features["pod_hair"] = pick(GLOB.pod_hair_list)
+	human.dna.species.randomize_active_underwear(human)
 
-	human.update_body(is_creating = TRUE)
+	for(var/datum/species/species_path as anything in subtypesof(/datum/species))
+		var/datum/species/new_species = new species_path
+		new_species.randomize_features(human)
```

Comment:
why aren't you just calling this directly on the dna's species?

Reply:
As I mentioned in the body, this would make it so that anyone that changes species would be forced into the only pseudo-random features set by initialize.dna and could cause problems both with spawners and with edge-cases such as someone changing species, being randomized and then changing back, since they features of their original species would remain unchanged.
I don't think it's a big deal processing-wise because the parent is just a return so for 60 or so of those species (which are mostly golems) so we're not really using up that many resources.